File: FishBunDemo/src/main/java/com/sangcomz/fishbundemo/SubFragment.java
package com.sangcomz.fishbundemo;


import android.content.Intent;
import android.graphics.Color;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;

import com.sangcomz.fishbun.FishBun;
import com.sangcomz.fishbun.define.Define;

import java.util.ArrayList;


/**
 * A simple {@link Fragment} subclass.
 */
public class SubFragment extends Fragment {

    ArrayList<Uri> path = new ArrayList<>();
    ImageView imgMain;
    Button btnAddImages;
    RecyclerView recyclerView;
    LinearLayoutManager linearLayoutManager;
    ImageAdapter imageAdapter;
    ImageController withActivityController;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_sub, container, false);
        // Inflate the layout for this fragment
        imgMain = (ImageView) rootView.findViewById(R.id.img_main);
        recyclerView = (RecyclerView) rootView.findViewById(R.id.recyclerview);
        btnAddImages = (Button) rootView.findViewById(R.id.btn_add_images);
        linearLayoutManager = new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false);
        withActivityController = new ImageController(getActivity(), imgMain);
        imageAdapter = new ImageAdapter(getActivity(), withActivityController, path);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setAdapter(imageAdapter);


        btnAddImages.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FishBun.with(SubFragment.this)
                        .MultiPageMode()
                        .setPickerCount(10)
                        .setActionBarColor(Color.parseColor("#3F51B5"), Color.parseColor("#303F9F"))
                        .setArrayPaths(path)
                        .setCamera(true)
                        .startAlbum();
            }
        });

        return rootView;
    }


    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        switch (requestCode) {
            case Define.ALBUM_REQUEST_CODE:
                if (resultCode == getActivity().RESULT_OK) {
                    path = data.getParcelableArrayListExtra(Define.INTENT_PATH);
                    imageAdapter.changePath(path);
                    break;
                }
        }
    }
}

File: FishBunDemo/src/main/java/com/sangcomz/fishbundemo/SubFragment.kt
package com.sangcomz.fishbundemo

import android.app.Activity
import android.content.Intent
import android.graphics.Color
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.sangcomz.fishbun.FishBun
import com.sangcomz.fishbun.adapter.image.impl.CoilAdapter
import com.sangcomz.fishbundemo.databinding.FragmentSubBinding

/**
 * A simple [Fragment] subclass.
 */
class SubFragment : Fragment() {

    var path = ArrayList<Uri>()
    private lateinit var imageAdapter: ImageAdapter
    private var _binding: FragmentSubBinding? = null
    private val binding get() = _binding!!

    private val startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
        if (it.resultCode == AppCompatActivity.RESULT_OK) {
            path = it.data?.getParcelableArrayListExtra(FishBun.INTENT_PATH) ?: arrayListOf()
            imageAdapter.changePath(path)
        }
    }


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentSubBinding.inflate(inflater, container, false)
        return binding.root
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        with(binding.recyclerview) {
            layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
            imageAdapter = ImageAdapter(requireActivity(), ImageController(binding.imgMain), path)
            adapter = imageAdapter
        }

        binding.btnAddImages.setOnClickListener {
            FishBun.with(this@SubFragment)
                .setImageAdapter(CoilAdapter())
                .setMaxCount(10)
                .setActionBarColor(Color.parseColor("#3F51B5"), Color.parseColor("#303F9F"))
                .setSelectedImages(path)
                .hasCameraInPickerPage(true)
                .startAlbumWithOnActivityResult(IMAGE_PICKER_REQUEST_CODE)
        }

        binding.btnAddImagesUseCallback.setOnClickListener {
            FishBun.with(this@SubFragment)
                .setImageAdapter(CoilAdapter())
                .setMaxCount(10)
                .setActionBarColor(Color.parseColor("#3F51B5"), Color.parseColor("#303F9F"))
                .setSelectedImages(path)
                .hasCameraInPickerPage(true)
                .startAlbumWithActivityResultCallback(startForResult)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            IMAGE_PICKER_REQUEST_CODE -> if (resultCode == Activity.RESULT_OK) {
                path = data?.getParcelableArrayListExtra(FishBun.INTENT_PATH) ?: arrayListOf()
                imageAdapter.changePath(path)
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        const val IMAGE_PICKER_REQUEST_CODE = 1
        fun newInstance() = SubFragment()
    }
}